packages/frontend/apps/electron/src/helper/workspace/handlers.ts

```diff
@@ -1,27 +1,63 @@
 import path from 'node:path';
 
+import { parseUniversalId } from '@affine/nbstore';
 import fs from 'fs-extra';
 
 import { logger } from '../logger';
+import { getDocStoragePool } from '../nbstore';
 import { ensureSQLiteDB } from '../nbstore/v1/ensure-db';
 import type { WorkspaceMeta } from '../type';
 import {
   getDeletedWorkspacesBasePath,
+  getSpaceDBPath,
   getWorkspaceBasePathV1,
   getWorkspaceMeta,
 } from './meta';
 
-export async function deleteWorkspace(id: string) {
+/**
+ * Permanently delete the workspace data
+ */
+export async function deleteWorkspace(universalId: string) {
+  const { peer, type, id } = parseUniversalId(universalId);
+  const dbPath = await getSpaceDBPath(peer, type, id);
   const basePath = await getWorkspaceBasePathV1('workspace', id);
+
+  try {
+    await getDocStoragePool().disconnect(universalId);
+  } catch {
+    // incase any connection existing
+  }
+
+  try {
+    await fs.rmdir(path.dirname(dbPath), { recursive: true });
+    await fs.rmdir(basePath, { recursive: true });
+  } catch (e) {
+    logger.error('deleteWorkspace', e);
+  }
+}
+
+/**
+ * Move the workspace folder to `deleted-workspaces`
+ * At the same time, permanently delete the v1 workspace folder if it's id exists in nbstore,
+ * because trashing always happens after full sync from v1 to nbstore.
+ */
+export async function trashWorkspace(universalId: string) {
+  const { peer, type, id } = parseUniversalId(universalId);
+  const dbPath = await getSpaceDBPath(peer, type, id);
+
   const movedPath = path.join(await getDeletedWorkspacesBasePath(), `${id}`);
   try {
     const db = await ensureSQLiteDB('workspace', id);
-    await db.destroy();
-    return await fs.move(basePath, movedPath, {
+    if (db) {
+      await db.destroy();
+      await fs.rmdir(path.dirname(db.path));
```

Comment:
add `{ recursive: true }` option

Reply:
recursive option got deprecated on `rmdir`

Reply:
I see. I docs says the alternative is `await fs.rm(path.dirname(db.path), { recursive: true });`